src/org/labkey/test/tests/issues/IssueAPITest.java

```diff
@@ -0,0 +1,492 @@
+        for (Long issueId : issueIds)
+        {
+            var issueresponseModel = getIssueResponse(issueId);
+            switch (issueresponseModel.getTitle())
+            {
+                case "first of three":
+                {
+                    assertEquals("expect new status", "resolved", issueresponseModel.getStatus());
+                    assertEquals("expect fixed", "fixed", issueresponseModel.getResolution());
+                    break;
+                }
+                case "second of three":
+                {
+                    assertEquals("expect new status", "closed", issueresponseModel.getStatus());
+                    break;
+                }
+                case "third of three":
+                {
+                    assertEquals("expect assignment to buddy",
+                            TEST_BUDDY_ID, issueresponseModel.getAssignedTo());
+                    break;
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testWithNoAction() throws Exception
+    {
+        IssueModel issue = new IssueModel()
+                .setTitle("test with no action")
+                .setComment("hopefully blows up")
+                .setAssignedTo(TEST_USER_ID)
+                .setNotify(TEST_BUDDY_NAME)
+                .setPriority(3L)
+                .setType("Defect");
+        try
+        {
+            doIssueAction(issue);
+            fail("should produce an error");
+        }catch (CommandException success)
+        {
+            assertEquals("Action must be specified : (update, resolve, close, reopen)", success.getMessage());
+        }
+    }
+
+    @Test
+    public void testIssueWithoutAssignedTo() throws Exception
+    {
+        IssueModel issue = new IssueModel()
+                .setTitle("test with no assignedTo")
+                .setComment("should assign to guest")
+                .setAction(IssueModel.IssueAction.insert)
+                .setNotify(TEST_BUDDY_NAME)
+                .setPriority(3L)
+                .setType("Defect");
+
+        var issueResponse= getIssueResponse( doIssueAction(issue));
+        assertEquals("should be assigned to guest", Long.valueOf(0), issueResponse.getAssignedTo());
+        assertEquals("should be open", "open", issueResponse.getStatus());
+    }
+
+    @Test
+    public void testIssueWithoutPri() throws Exception
+    {
+        IssueModel issue = new IssueModel()
+                .setTitle("test with no priority")
+                .setComment("should get a CommandException")
+                .setAction(IssueModel.IssueAction.insert)
+                .setType("Defect");
+
+        try{
+            doIssueAction(issue);
+            fail("expect error if command is executed without required property priority");
+        }catch(CommandException success)
+        {
+            assertThat(success.getMessage(), containsString("Data does not contain required field: priority"));
+            resetErrors();
```

Comment:
I wouldn't expect a missing field to log a server error. This seems like a bug.
@labkey-klum ?

Reply:
Correct, the server should respond with a 400 not throw. It looks like it's not catching this in validation but later during the commit. Probably has something to do with the difference between a UI and API post.

If you posted this instead : `priority : null` you would probably see a different error.

@labkey-chrisj I would open an issue and we can remove this code once we shore up validation.

Reply:
OK, I've opened https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=44784
I'll add a note in the test code along with other feedback